### PR TITLE
Refactor the key forwarding logic and tests

### PR DESCRIPTION
**Pull request by poljar**
No changes in the logic, just removing a bit of code duplication in the tests and splitting out some methods.

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/gossiping/machine.rs [hunk 1]
```diff
@@ -1039,11 +1037,11 @@ mod tests {
         ReadOnlyAccount::new(alice_id(), alice2_device_id())
     }
 
-    fn bob_machine() -> GossipMachine {
-        let user_id = Arc::from(bob_id());
-        let account = ReadOnlyAccount::new(&user_id, alice_device_id());
+    fn bob_machine(user_id: &UserId) -> GossipMachine {
+        let user_id = Arc::from(user_id);
+        let account = ReadOnlyAccount::new(&user_id, bob_device_id());
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/gossiping/machine.rs:1042]: I find this pretty confusing, maybe because I am not familiar with the API being tested. When I see `ReadOnlyAccount::new(user_id, device_id)`, it looks to me as if the device ID would naturally be of a device for that user, but here they seem to be mostly unrelated?

Comment (poljar) [reply crates/matrix-sdk-crypto/src/gossiping/machine.rs:1042]: The only requirement here is that the device id is unique, we don't use more than two device ids in the whole test module so that requirement is met.

Perhaps it would be better if we generate a random device id?

Comment (jplatte) [reply crates/matrix-sdk-crypto/src/gossiping/machine.rs:1042]: If that makes the test work the same way, it seems preferable to me.

Comment (poljar) [reply crates/matrix-sdk-crypto/src/gossiping/machine.rs]: Not complicated at all: https://github.com/matrix-org/matrix-rust-sdk/pull/901/commits/1d27d72c8f1f9d7e040acfe9e5b6af36ad6a2e62.

File: crates/matrix-sdk-crypto/src/gossiping/machine.rs [hunk 2]
```diff
@@ -1032,18 +1037,20 @@ mod tests {
         ReadOnlyAccount::new(alice_id(), alice2_device_id())
     }
 
-    fn bob_machine() -> GossipMachine {
-        let user_id = Arc::from(bob_id());
-        let account = ReadOnlyAccount::new(&user_id, alice_device_id());
+    fn bob_machine(user_id: &UserId) -> GossipMachine {
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/gossiping/machine.rs:1040]: Okay, now this I'm pretty sure this function name no longer makes sense 😅 

Maybe `test_gossip_machine()` or sth. like that?

**Review by jplatte — APPROVED** [review]
One more nit, otherwise LGTM.

File: crates/matrix-sdk-crypto/src/gossiping/machine.rs (hunk 2 above)
Comment (poljar) [reply crates/matrix-sdk-crypto/src/gossiping/machine.rs]: Right, there we go https://github.com/matrix-org/matrix-rust-sdk/pull/901/commits/a5800080b802b2604c72ee37cf9ccf1bbf5ce0c3.
